### PR TITLE
Gradle deps

### PR DESCRIPTION
**Pull request by stevemessick**
@devoncarew 

Not quite sure how this will be used yet. It takes a while to run, so we'll need to design around that. I'd say it is comparable to analyzing a Dart project.

### CODE REVIEW

File: flutter-studio/testSrc/io/flutter/GradleDependencyFetcherTest.java [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+
+  @BeforeClass
+  public static void setUp() {
+    Disposable disposable = Disposer.newDisposable();
+    ApplicationManager.setApplication(new MockApplication(disposable), disposable);
+    Extensions.registerAreaClass("IDEA_PROJECT", null);
+    ourProject = new MockProject(ApplicationManager.getApplication().getPicoContainer(), disposable);
+  }
+
+  private static String getDependencyReport() {
```
Comment (devoncarew) [inline flutter-studio/testSrc/io/flutter/GradleDependencyFetcherTest.java]: `createMockDependencyReport()`?

File: flutter-studio/src/io/flutter/android/GradleDependencyFetcher.java [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+    myProject = project;
+    myDependencies = new HashMap<>(15);
+  }
+
+  @NotNull
+  public Map<String, List<String>> getDependencies() {
+    return myDependencies;
+  }
+
+  public void run() {
```
Comment (devoncarew) [inline flutter-studio/src/io/flutter/android/GradleDependencyFetcher.java]: When this is used, I assume the result will either be available synchronously, or asynchronously via a Future?

File: flutter-studio/testSrc/io/flutter/GradleDependencyFetcherTest.java (hunk 1 above)
Comment (stevemessick) [reply flutter-studio/testSrc/io/flutter/GradleDependencyFetcherTest.java]: OK

File: flutter-studio/src/io/flutter/android/GradleDependencyFetcher.java (hunk 2 above)
Comment (stevemessick) [reply flutter-studio/src/io/flutter/android/GradleDependencyFetcher.java]: Well, run() is synchronous. But I expect we'll have something driving the process that is async and also takes care of the progress UI.

**Pull request merged**